demos/shape/shape.html

```diff
@@ -0,0 +1,104 @@
+    <script src="/assets/shape/shape.css.js"></script>
+  </head>
+  <body>
+    <header class="mdc-toolbar mdc-toolbar--fixed">
+      <div class="mdc-toolbar__row">
+        <section class="mdc-toolbar__section mdc-toolbar__section--align-start">
+          <span class="catalog-back">
+            <a href="/" class="mdc-toolbar__icon--menu"><i class="material-icons">&#xE5C4;</i></a>
+          </span>
+          <span class="mdc-toolbar__title catalog-title">Shape</span>
+        </section>
+      </div>
+    </header>
+    <main>
+      <div class="mdc-toolbar-fixed-adjust"></div>
+      <section class="hero">
+          <div class="mdc-shape">
+            <canvas class="mdc-shape__canvas"></canvas>
+            <svg class="mdc-shape__svg">
+              <clipPath>
+                <path class="mdc-shape__path"/>
+              </clipPath>
+            </svg>
+          </div>
+      </section>
+      <section>
+        <fieldset>
+          <legend class="mdc-typography--title">Change Elevation</legend>
+          <figure>
+            <div class="mdc-shape" id="elevatable-shape">
+              <canvas class="mdc-shape__canvas"></canvas>
+              <svg class="mdc-shape__svg">
+                <clipPath>
+                  <path class="mdc-shape__path"/>
+                </clipPath>
+              </svg>
+            </div>
+          </figure>
+          <figcaption class="demo-input">
+            <span>Elevation</span>
+            <input type="number" name="elevation-input" value="4" id="elevation-input">
+          </figcaption>
+        </fieldset>
+      </section>
+    </main>
+    <script src="/assets/material-components-web.js"></script>
+    <script src="/assets/shape/heart/index.js"></script>
+    <script>
+      // Because we load our CSS via webpack, we need to ensure that all of the correct styles
+      // are applied before ripples are attached. Otherwise, ripples may use the computed styles of
+      // elements before our CSS is applied, leading to improper UX.
+      (function() {
+        var pollId = 0;
+        pollId = setInterval(function() {
+          var pos = getComputedStyle(document.querySelector('.mdc-shape')).position;
+          if (pos === 'relative') {
+            init();
+            clearInterval(pollId);
+          }
+        }, 250);
+        function init() {
+          const DemoHeart = mdc["shape/heart/index"].default;
+          var shapes = document.querySelectorAll(".mdc-shape");
+          let elevatableHeart;
+          for (var i=0; i<shapes.length; i++) {
+            const demoHeart = DemoHeart.attachTo(shapes[i]);
+            if (shapes[i].id === "elevatable-shape") {
+              elevatableHeart = demoHeart;
+            }
+          }
+          const elevationInput = document.querySelector("#elevation-input");
+          elevationInput.addEventListener("change", () => {
+            elevatableHeart.elevation = elevationInput.value;
```

Comment:
I'm noticing that there is sometimes visible jank when changing this elevation and I'm not sure whether it's a flicker of the clip-path resetting or what.

Reply:
There was a bug with parsing the elevation value into an int, this was causing the penumbra shadow to draw in the wrong place. Fixed the bug and now I see now jank.

FYI, the clip-path doesn't change when the elevation changes .... but the entire canvas redraws. Its not GPU accelerated, but thats kinda the deal with shapes.